### PR TITLE
Add chip classification backend and example experiment

### PR DESCRIPTION
**Pull request by AdeelH**
The output of the full experiment run are available at: https://s3.console.aws.amazon.com/s3/buckets/raster-vision-ahassan/fastai/spacenet/rio/remote-output/?region=us-east-1

The eval metrics are about the same as for the Keras backend:
```{JSON}
[
    {
        "precision": 0.9610891701370387,
        "recall": 0.9739904175222451,
        "f1": 0.9671508020220596,
        "count_error": 0.0,
        "gt_count": 1461.0,
        "class_id": 1,
        "class_name": "building"
    },
    {
        "precision": 0.9834804892287192,
        "recall": 0.9740596627756162,
        "f1": 0.9787069337794698,
        "count_error": 0.0,
        "gt_count": 2313.0,
        "class_id": 2,
        "class_name": "no_building"
    },
    {
        "precision": 0.9748123076725599,
        "recall": 0.9740328563857976,
        "f1": 0.9742332961277538,
        "count_error": 0.0,
        "gt_count": 3774.0,
        "class_id": null,
        "class_name": "average"
    }
]
```

### CODE REVIEW

File: fastai_plugin/chip_classification_backend.py [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+        print('--------------')
+        for k, v in self.train_opts.__dict__.items():
+            print('{}: {}'.format(k, v))
+        print()
+
+    def process_scene_data(self, scene, data, tmp_dir):
+        """Process each scene's training data.
+
+        This writes {scene_id}/img/{scene_id}-{ind}.png and
+        {scene_id}/labels/{scene_id}-{ind}.png
```
Comment (lewfish) [inline fastai_plugin/chip_classification_backend.py]: This comment should be corrected.

File: fastai_plugin/chip_classification_backend.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+
+    def process_sceneset_results(self, training_results, validation_results, tmp_dir):
+        """After all scenes have been processed, process the result set.
+
+        This writes a zip file for a group of scenes at {chip_uri}/{uuid}.zip
+        containing:
+        train-img/{scene_id}-{ind}.png
+        train-labels/{scene_id}-{ind}.png
+        val-img/{scene_id}-{ind}.png
+        val-labels/{scene_id}-{ind}.png
```
Comment (lewfish) [inline fastai_plugin/chip_classification_backend.py]: Same thing -- update comment.

**Comment by lewfish** [comment]
Also, as a general rule, which can be broken sometimes, you should have a short description of what you did in the PR description and what you did to test it.

**Comment by AdeelH** [comment]
Comments fixed. 

The output of the full experiment run are available at: https://s3.console.aws.amazon.com/s3/buckets/raster-vision-ahassan/fastai/?region=us-east-1

The eval metrics are about the same as for the Keras backend:
```{JSON}
[
    {
        "precision": 0.9610891701370387,
        "recall": 0.9739904175222451,
        "f1": 0.9671508020220596,
        "count_error": 0.0,
        "gt_count": 1461.0,
        "class_id": 1,
        "class_name": "building"
    },
    {
        "precision": 0.9834804892287192,
        "recall": 0.9740596627756162,
        "f1": 0.9787069337794698,
        "count_error": 0.0,
        "gt_count": 2313.0,
        "class_id": 2,
        "class_name": "no_building"
    },
    {
        "precision": 0.9748123076725599,
        "recall": 0.9740328563857976,
        "f1": 0.9742332961277538,
        "count_error": 0.0,
        "gt_count": 3774.0,
        "class_id": null,
        "class_name": "average"
    }
]
```

**Comment by lewfish** [comment]
After a PR is approved, the convention is for the author to merge it.

**Comment by AdeelH** [comment]
I don't think I have the necessary permissions

**Comment by lewfish** [comment]
I just gave you write access for this and https://github.com/azavea/raster-vision

**Pull request merged**